### ISSUE TITLE
Bug: columns are not becomes camelCase at 1.1.0

### ISSUE THREAD

**Issue by felixmosh** [Kequc/knex-stringcase#7]
Hi,

I've updated to the latest version (1.1.0), and now the columns names are not converting to camelCase at all.

I've debugged it a bit, looks like this [line](https://github.com/Kequc/knex-stringcase/blob/master/index.js#L63) always `false` cause the object that knex returns has **TextRow** as a constructor.

![image](https://user-images.githubusercontent.com/9304194/54536418-5ff18b00-4999-11e9-8150-717aa41f5e54.png)

Looks like this data type is what mysql2 returns, therefore the check for constructor is invalid.

**Comment by Kequc** [comment]
Ever since supporting deep data structures I've been struggling with detection of whether or not an item coming back from knex is an object or a value. As in your example a special exception needs to be put in place for `Date` which is a solution I used too. Then it turns out some people were having knex return dates generated by the `moment` library.

I thought I really had a solution in `1.1.0` I didn't consider `mysql2` or any other libraries might be formatting their results sets with something other than plain objects. I need something that can tell the difference between Date/Moment/etc, and Object/TextRow/etc.

As they fundamentally need to be treated differently.

The linked solution unfortunately just takes us in a circle, where the next ticket that comes in will say `moment` formatted dates no longer work anymore. And even with `Date` and `moment` being special exceptions, there will be some other strange format later.

What is the best ultimate solution here?

**Comment by felixmosh** [comment]
@Kequc, first of all thank you for the fast response 👏🏽

I think that we should tackle this issue by cases, if moment is the problematic, let's write a test for it, and handle it.

But we we must support basic mysql client (mysql2).

**Comment by Kequc** [comment]
I'm thinking since there's a limited number of libraries you can use with knex, but an unlimited number of ways to format the values that come back. This library should detect (from the user's config) what lib they are using (ie. `mysql2`) and then expect to see rows formatted by that library.

For example if it detects `mysql2` then it expects rows to look like `TextRow` the trick will be creating an index of the different libraries and their return formats.

**Comment by felixmosh** [comment]
Ham, It is a possibility.
I think that we can just go with a black list approach (hardcoded), 
Check for `.constructor === Object` || `.constructor.name === 'TextRow' || ... rest allowed types`.
The problem is that we won't be able to know when that class name will be renamed.

An alternative solution can be to open an issue to MySql2 to prevent exposing internal objects outside.

**Comment by Kequc** [comment]
My hesitation about this solution is that I'm not sure if TextRow is always what `mysql2` will return... are there other formats it might send back?

What about other libraries what to they return... right?

I agree these need to be hardcoded.

**Comment by felixmosh** [comment]
Look, u didn't had issues regarding the previous condition, maybe there is no need for more strict check? 😄

**Comment by Kequc** [comment]
People were returning custom types for their dates and could return custom types for anything. That needs to be supported because it's supported by knex.

**Comment by felixmosh** [comment]
Can u share an example, I'm not familiar with it.

**Comment by Kequc** [comment]
https://github.com/Kequc/knex-stringcase/issues/6#issuecomment-471568322

You can see here.

**Comment by felixmosh** [comment]
> People were returning custom types for their dates and could return custom types for anything. That needs to be supported because it's supported by knex.

Where I can see a usage / config of custom types in knex?

**Comment by felixmosh** [comment]
Oh, I think that I found something... 
```javascript
TextRow.toString() === "[object Object]"
({}).toString() === "[object Object]"
(moment()).toString() === ""
(new Date()).toString() === "Mon Mar 18 2019 21:52:45 GMT+0200 (Israel Standard Time)"
(moment()).toString() === "Mon Mar 18 2019 21:53:18 GMT+0200"
```

**Comment by Kequc** [comment]
That's a good hint! Unfortunately I don't think it's robust :( because `"[object Object]"` could be almost anything in javascript unfortunately.

I've released `1.1.1` which goes back to non strict row detection, because the problem you are having is really important and needs to be fixed right away.

At the same time I realised if people can manipulate what values come back from knex, they can also manipulate what row types come back. Therefore I've decided it's up to them to specify explicitly when they're doing that kind of stuff. So I've re-introduced the `ignoreStringcase` config option for them to use.

This is the best solution. Because it means anyone can use this library for anything, as long as they're willing to config. Thanks for helping me and talking through it with me I think you gave me some valuable insights and got me to think about it more.
